requests-for-comments/2025-06-18-sdk-specs.md

```diff
@@ -0,0 +1,301 @@
+│  ║ Create a new user account     ║ │
+│  ║                               ║ │
+│  ║ Parameters:                   ║ │
+│  ║   username (string)           ║ │
+│  ║     User name to create       ║ │
+│  ║                               ║ │
+│  ║ Returns:                      ║ │
+│  ║   User - Created user object  ║ │
+│  ║                               ║ │
+│  ║ [Python] [JavaScript] [Node]  ║ │
+│  ║                               ║ │
+│  ║ try:                          ║ │
+│  ║   user = create_user("alice") ║ │
+│  ║ except Exception as e:        ║ │
+│  ║   print(f"Error: {e}")        ║ │
+│  ╚═══════════════════════════════╝ │
+└─────────────────────────────────────┘
+```
+
+### Doc comments
+We'll try to use established doc comment formats where possible. For example, in the PostHog Web JavaScript, we can use [Microsoft's `tsdoc`](https://tsdoc.org/) as a relatively established and well-tooled doc comment format.
+
+The doc comments will cover:
+- Brief description of the method's purpose
+- Description of parameters
+- Remarks for additional details about a function's usage
+- Inline examples of usage
+- Category tags, like initialization, persons, feature flags, etc.
+- Return signature description where applicable
+
+Example
+```ts
+// Initialization methods
+
+    /**
+     * Initializes a new instance of the PostHog capturing object.
+     * 
+     * @remarks
+     * All new instances are added to the main posthog object as sub properties (such as
+     * `posthog.library_name`) and also returned by this function. [Learn more about configuration options](/docs/libraries/js/config)
+     *
+     * @example
+     * ```js
+     * // basic initialization
+     * posthog.init(
+     * '<ph_project_api_key>', { 
+     *     api_host: '<ph_client_api_host>', 
+     *     defaults: '<ph_posthog_js_defaults>' 
+     * })
+     * ```
+     * 
+     * @example
+     * ```js
+     * // multiple instances
+     * posthog.init('<ph_project_api_key>', {}, 'project1')
+     * posthog.init('<ph_project_api_key>', {}, 'project2')
+     * ```
+     * 
+     * @public
+     *
+     * @param token - Your PostHog API token
+     * @param config - A dictionary of config options to override. See [default config options](https://github.com/posthog/posthog-js/blob/6e0e873/src/posthog-core.js#L57-L91)
+     * @param name - The name for the new posthog instance that you want created
+     * 
+     * {@label Initialization}
+     * 
+     * @returns The newly initialized PostHog instance
+     */
+```
+
+This is **very similar** to what we already do, but standardizing this to be parsable will go a long way. This also ensures that when SDK code is updated, so will the specs be updated.
+
+### Specs generated
+
+There isn't a good standard I like. So we'll create `hogRef` inspired by [Supabase's SDK refs](https://raw.githubusercontent.com/supabase/supabase/82715efc250bd1e706491effab91968ea100479a/apps/docs/spec/supabase_dart_v2.yml).
```

Comment:
Is it likely other people will want to consume this `hogRef` file or is this internal-only for our website? We'd probably want to do something more mainstream if this is supposed to be consumed by our AI overlords, for example

Reply:
I'm open to other spec suggestions, but I actually didn't find a good, language agnostic spec that's available. 

I would love to not have to invent our own thing, but I did this at Appwrite, Supabase is doing their own thing, afaik, Stripe is doing their own thing, too.